earth_enterprise/src/fusion/asset-tests/bin/generate_test_asset_xmls.py

```diff
@@ -9,13 +9,73 @@
 file operations in the test assets directory slow.
 """
 
+import argparse
+import functools
 import jinja2
+import math
 import os
 import os.path
 import sys
 import uuid
 
 
+# Create some decorators for methods that serve as tasks to be executed.
+def execute_once(func):
+    """Make a function exectule only once.  Calls after the first one return
+    immediately.
+    """
+
+    @functools.wraps(func)
+    def wrapper_execute_once(*args, **kwargs):
+        if wrapper_execute_once.executed:
+            return
+
+        res = func(*args, **kwargs)
+        wrapper_execute_once.executed = True
+
+        return res
+
+    wrapper_execute_once.executed = False
+
+    return wrapper_execute_once
+
+
+class ProgressMeter(object):
+    """A basic interface for reporting task progress."""
+
+    def __init__(self, start_value, end_value, update_interval=1000):
+        """
+            update_interval = How much the progress value should change before
+        the display is updated.
+        """
+
+        self.start_value = start_value
+        self.end_value = end_value
+        self.value = start_value
+        self.update_interval = update_interval
+        self.display_last_value = None
+
+    def advance(self, progress_amount):
+        self.value += progress_amount
+        self.update_progress()
+
+    def close(self):
+        sys.stdout.write(os.linesep)
+
+    def update_progress(self):
+        if (self.value - self.start_value) % self.update_interval == 0:
```

Comment:
Minor: if `progress_amount` passed to `advance` is not 1, this may skip printing progress sometimes.